roles/etcd/templates/etcdctl.sh.j2

```diff
@@ -3,19 +3,20 @@
 # command flags are different between the two. Should work on stand
 # alone etcd hosts and master + etcd hosts too because we use the peer keys.
 etcdctl2() {
- cmd="/usr/bin/etcdctl --cert-file {{ etcd_peer_cert_file }} --key-file {{ etcd_peer_key_file }} --ca-file {{ etcd_peer_ca_file }} -C https://`hostname`:2379 ${@}"
+
+ cmd="ETCDCTL_API=2 etcdctl --cert-file {{ etcd_peer_cert_file }} --key-file {{ etcd_peer_key_file }} --ca-file {{ etcd_peer_ca_file }} -C https://`hostname`:2379 ${@}"
  if [[ -f /usr/local/bin/master-exec ]]; then
-   /usr/local/bin/master-exec etcd etcd /bin/bash -c "$cmd"
+   /usr/local/bin/master-exec etcd etcd /bin/sh -c "$cmd"
```

Comment:
I believe this is busybox shell in coreos image.  It accepts -c, so this should work fine.